src/coreclr/src/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.targets

```diff
@@ -144,6 +144,37 @@ Copyright (c) .NET Foundation. All rights reserved.
     <CallTarget Targets="$(TargetsTriggeredByCompilation)" Condition="'$(TargetsTriggeredByCompilation)' != ''"/>
   </Target>
 
+  <Target Name="GenerateVersionFile">
+   <PropertyGroup>
+      <VersionFilePath Condition="'$(VersionFilePath)' == ''">$([MSBuild]::NormalizePath('$(IntermediateOutputPath)', 'version', 'version.h'))</VersionFilePath>
+      <_DotnetVersion>$(AssemblyVersion.Replace('.', ':'))</_DotnetVersion>
```

Comment:
Does it really make sense to tie DotnetVersion and AssemblyVersion? These versions are not same even for the case we are using it for System.Runtime.CompilerServices.Unsafe, e.g.:

netstandard2.0 flavor should have System.Runtime 4:0:0:0 here, but the assembly version is 5.0.0.0.

Reply:
It would be useful to show the matching testeed changes in System.Runtime.CompilerServices.Unsafe to take advantage of this.

I am having doubts whether this feature belongs to the IL SDK at all.

Reply:
https://github.com/dotnet/runtime/pull/39670  this contains the matching changes. (The target can be removed after the pr is merged.)

Reply:
I think the ILSDK should support some form of MSBuild-sourced assembly and file versions (as well as other assembly metadata) so that its consistent with the .NETSDK WRT versioning and we don't have to "remember" to check in versioning changes to .il files every release.

Is it possible to produce a self-contained '.il' file rather than a header that the checked-in source needs to know about?

Reply:
I tried generating a IL file with the version and adding to the compile
```
.assembly System.Runtime.CompilerServices.Unsafe
{
  .ver ASSEMBLY_VERSION
}
```

But compiling doesn't produce the correct version in the assembly, It seems like we need to define all the assembly attributes together.

Reply:
> It seems like we need to define all the assembly attributes together.

I see, indeed after playing with using il files it looks like ilasm isn't really built to support many source files to one output file.  I also tried having multiple `.assembly` directives with same name to test for instance, what it would look like if one came from the header, and ilasm silently used only the first one, so that's not really an option either.  We have to define the entire `.assembly` directive in one place.